src/components/accordion/accordion.js

```diff
@@ -0,0 +1,169 @@
+  // Check module exists
+  var $module = this.$module
+  if (!$module) {
+    return
+  }
+
+  this.moduleId = $module.getAttribute('id')
+
+  nodeListForEach(this.$sections, function ($section, i) {
+    // Set header attributes
+    var header = $section.querySelector('.govuk-accordion__section-header')
+    this.setHeaderAttributes(header, i)
+
+    this.setExpanded(this.isExpanded($section), $section)
+
+    // Handle events
+    header.addEventListener('keypress', this.onKeyPressed.bind(this, $section))
+    header.addEventListener('click', this.onToggleExpanded.bind(this, $section))
+  }.bind(this))
+
+  // Create "Open all" button and set attributes
+  this.$openAllButton = document.createElement('button')
+  this.setOpenAllButtonAttributes(this.$openAllButton)
+
+  // Create controls and set attributes
+  var accordionControls = document.createElement('div')
+  accordionControls.setAttribute('class', 'govuk-accordion__controls')
+  accordionControls.appendChild(this.$openAllButton)
+  this.$module.insertBefore(accordionControls, this.$module.firstChild)
+
+  this.$module.classList.add('with-js')
+
+  // Handle events
+  this.$openAllButton.addEventListener('click', this.openOrCloseAllSections.bind(this))
+
+  // See if OpenAll button text should be updated
+  var areAllSectionsOpen = this.checkIfAllSectionsOpen()
+  this.updateOpenAllButton(areAllSectionsOpen)
+}
+
+// Open/close section
+Accordion.prototype.onToggleExpanded = function ($section) {
+  var expanded = this.isExpanded($section)
+  this.setExpanded(!expanded, $section)
+
+  // See if OpenAll button text should be updated
+  var areAllSectionsOpen = this.checkIfAllSectionsOpen()
+  this.updateOpenAllButton(areAllSectionsOpen)
+}
+
+Accordion.prototype.onKeyPressed = function (section, event) {
+  if (event.keyCode === KEY_ENTER || event.keyCode === KEY_SPACE) {
+    event.preventDefault()
+
+    // Open/close section
+    this.onToggleExpanded(section)
+  }
+}
+
+// Toggle aria-expanded when section opened/closed
+Accordion.prototype.setExpanded = function (expanded, $section) {
+  var $button = $section.querySelector('.govuk-accordion__section-header-button')
+  $button.setAttribute('aria-expanded', expanded)
+
+  if (expanded) {
+    $section.classList.add('govuk-accordion__section--expanded')
+  } else {
+    $section.classList.remove('govuk-accordion__section--expanded')
+  }
+
+  // This is set to trigger reflow for IE8, which doesn't
+  // always reflow after a setAttribute call.
+  this.$module.className = this.$module.className
+}
+
+Accordion.prototype.isExpanded = function ($section) {
+  return ($section.classList.contains('govuk-accordion__section--expanded'))
+}
+
+Accordion.prototype.setHeaderAttributes = function ($header, index) {
+  $header.setAttribute('tabindex', '0')
+
+  var $button = $header.querySelector('.govuk-accordion__section-header-button')
+  $button.setAttribute('aria-controls', this.moduleId + '-panel-' + index)
```

Comment:
`index` now needs to start at 1 following the change from https://github.com/alphagov/govuk-frontend/pull/958#pullrequestreview-169372580. Apologies I didn't realise the change would affect this.